SAMv1.tex

```diff
@@ -500,11 +500,36 @@ \subsection{The alignment section: mandatory fields}\label{sec:alnrecord}
   {\sf RNEXT} and bit 0x20.
 \item {\sf TLEN}: signed observed Template LENgth. If all segments are
   mapped to the same reference, the unsigned observed template length
-  equals the number of bases from the leftmost mapped base to the
-  rightmost mapped base. The leftmost segment has a plus sign and the
-  rightmost has a minus sign. The sign of segments in the middle is
-  undefined. It is set as 0 for single-segment template or when the
-  information is unavailable.
+  equals the distance between the mapped end of the template and the
+  mapped start of the template, inclusively.\footnote{Thus a segment
+    aligning in the forward direction at base 100 for length 50 and a
+    segment aligning in the reverse direction at base 200 for length
+    50 indicate the template covers bases 100 to 249 and has length
+    150.}
+  Note that mapped base is defined to be one that aligns to the
+  reference as described by CIGAR, hence exclude soft-clipped bases.
+  The TLEN field is positive for leftmost segment of the template,
+  negative for the rightmost and the sign for any middle segment is
+  undefined.  If segments cover the same coordinates then the choice
+  of which is leftmost and rightmost is arbitrary, but the two ends
+  must still have differing sign.
+  It is set as 0 for single-segment template or when the
+  information is unavailable (e.g. for a single-segment template or
+  when the first or last segment of a multi-segment template is
+  unmapped or are mapped to different references).
+
+  The intention of this field is to indicate where the other end of
+  the template has been aligned without needing to read the remainder
+  of the SAM file.  Unfortunately there has been no
+  clear consensus on the definition the template mapped start
+  and end.  Thus the exact definitions are implementation defined.%
+\footnote{The earliest versions of this specification used $5'$ to
+  $5'$ (in original orientation) while later ones used leftmost to
+  rightmost.  Note: these two definitions agree in many alignments for
+  many paired library strategies, but differ in the case of overlaps
+  where the first segment aligns beyond the start of the last segment.
+  When implementing a new alignment algorithm it is recommended to
+  choose one of the above two definitions.}
```

Comment:
Maybe we can be more definitive and clearer than “choose one of the above” :smile:

Reply:
Maybe, but the whole point of the PR was to admit defeat that we've had years to make a choice and dismally failed to do so.  If we start to have the discussion on which to recommend I suspect we'll be here for another few years!  Unless everyone has since made their mind up in the same direction. :-)

Reply:
Similarly unclear is which is leftmost and which is rightmost...are we talking about the 5' ends? or the alignment POS?

Reply:
"leftmost mapped base".  How can that be unclear except in the case of ties, which I think we covered.

Fundamentally, we can't fix this.  It's been hanging around like a bad smell for *years*.  If it was easy to nail down to a single definition it'd have been done by now.  This PR is simply a pragmatic acceptance that we failed and there are (at least) two definitions out there already.  I'm open to better wording, but not open to ways of nailing it down to one meaning as that ship sadly sailed ages ago.